api/v1beta1/csiaddonsnode_types.go

```diff
@@ -0,0 +1,101 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type CSIAddonsNodeDriver struct {
+	// Name is the name of the CSI driver that this object refers to.
+	// This must be the same name returned by the CSI-Addons GetIdentity()
+	// call for that driver. The name of the driver is in the format:
+	// `rbd/cephfs.csi.ceph.com`
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+
+	// EndPoint is url that contains the ip-address to which the CSI-Addons
+	// side-car listens to.
+	EndPoint string
+}
+
+// CSIAddonsNodeState defines the state of the operation.
+type CSIAddonsNodeState string
+
+const (
+	// Connected represents the Connected state.
+	Connected CSIAddonsNodeState = "Connected"
+
+	// Failed represents the Connection Failed state.
+	Failed CSIAddonsNodeState = "Failed"
+)
+
+// CSIAddonsNodeSpec defines the desired state of CSIAddonsNode
+type CSIAddonsNodeSpec struct {
+	// Driver is the information of the CSI Driver existing on a node.
+	// If the driver is uninstalled, this can become empty.
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	Driver CSIAddonsNodeDriver `json:"driver" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,1,rep,name=drivers"`
+}
+
+// CSIAddonsNodeStatus defines the observed state of CSIAddonsNode
+type CSIAddonsNodeStatus struct {
+	// State represents the state of the CSIAddonsNode object.
+	// It informs whether or not the CSIAddonsNode is Connected
+	// to the CSI Driver.
+	State CSIAddonsNodeState `json:"status,omitempty"`
+
+	// Messgae is a human-readable message indicating details about why the CSIAddonsNode
+	// is in this state.
+	// +optional
+	Message string `json:"message,omitempty" protobuf:"bytes,2,opt,name=message"`
+
+	// Reason is a brief CamelCase string that describes any failure and is meant
+	// for machine parsing and tidy display in the CLI.
+	// +optional
+	Reason string `json:"reason,omitempty" protobuf:"bytes,3,opt,name=reason"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:printcolumn:JSONPath=".metadata.namespace",name=namespace,type=string
+//+kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",name=Age,type=date
+//+kubebuilder:printcolumn:JSONPath=".spec.driver.name",name=DriverName,type=string
+//+kubebuilder:printcolumn:JSONPath=".spec.driver.type",name=DriverType,type=string
+//+kubebuilder:printcolumn:JSONPath=".spec.driver.endpoint",name=Endpoint,type=string
+//+kubebuilder:printcolumn:JSONPath=".status.state",name=state,type=string
+//+kubebuilder:printcolumn:JSONPath=".status.message",name=message,type=string
+//+kubebuilder:printcolumn:JSONPath=".status.reason",name=reason,type=string
```

Comment:
the status should not be printed when someone does `oc get object` it's more for describe/yaml output